test/e2e/tests/vault-corruption/vault-corruption.spec.ts

```diff
@@ -0,0 +1,340 @@
+    browser.storage.local.set({ data: null }, () => {
+      ${code}
+    });
+`;
+
+  /**
+   * Common code to reload the extension; used by both the primary and backup
+   * database corruption scripts.
+   */
+  const reloadAndCallbackScript = `
+    browser.runtime.reload();
+    callback();
+  `;
+
+  /**
+   * Script to break the primary database only.
+   */
+  const breakPrimaryDatabaseOnlyScript = createCorruptionScript(
+    reloadAndCallbackScript,
+  );
+
+  /**
+   * Script to break both the primary and backup databases.
+   */
+  const breakAllDatabasesScript = createCorruptionScript(`
+    // indexedDB is not scuttled in test builds, so we can use it to access the
+    // backup database here
+    const request = globalThis.indexedDB.open('metamask-backup', 1);
+    request.onupgradeneeded = () => {
+      const db = request.result;
+      db.createObjectStore('store', { keyPath: 'id' });
+    };
+    request.onsuccess = () => {
+      const db = request.result;
+      const transaction = db.transaction('store', 'readwrite');
+      const store = transaction.objectStore('store');
+      // remove the KeyringController data from the backup database
+      store.delete('KeyringController');
+      transaction.oncomplete = () => {
+        ${reloadAndCallbackScript}
+      };
+    };`);
+
+  /**
+   * Returns the common config for these tests.
+   *
+   * @param title - The title of the test.
+   */
+  function getConfig(title?: string) {
+    return {
+      title,
+      ignoredConsoleErrors: [
+        // expected error caused by breaking the database:
+        'PersistenceError: Data error: storage.local does not contain vault data',
+      ],
+      // This flag ultimately requires that we onboard manually, as we can't use
+      // `fixtures` in this test, as the `ExtensionStore` class doesn't use them.
+      manifestFlags: {
+        testing: {
+          forceExtensionStore: true,
+        },
+      },
+    };
+  }
+
+  /**
+   * Onboard the user.
+   *
+   * @param driver - The WebDriver instance.
+   */
+  async function onboard(driver: Driver) {
+    return await completeCreateNewWalletOnboardingFlow({
+      driver,
+      password: WALLET_PASSWORD,
+      skipSRPBackup: true,
+    });
+  }
+
+  /**
+   * Since reloading the background restarts the extension the UI isn't
+   * available immediately. So we just keep reloading the UI until it is. This
+   * is a bit of a hack, but I can't figure out a better way.
+   *
+   * @param driver - The WebDriver instance.
+   */
+  async function waitForVaultRestorePage(driver: Driver) {
+    let title: string;
+    do {
+      await driver.navigate(PAGES.HOME, {
+        waitForControllers: false,
+      });
+      title = await driver.driver.getTitle();
+      // the browser will return an error message for our UI's HOME page until
+      // the extension has restarted
+    } while (title !== WINDOW_TITLES.ExtensionInFullScreenView);
```

Comment:
as Mark was suggesting in the call, maybe this could solve the while loop:

`await this.driver.wait(until.titleIs(WINDOW_TITLES.ExtensionInFullScreenView), timeout);`

wdyt?